### PR TITLE
Add Step Function client and utils

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -6,9 +6,10 @@ object Dependencies {
   lazy val lambdaJavaCore = "com.amazonaws" % "aws-lambda-java-core" % "1.2.1"
   lazy val lambdaJavaEvents = "com.amazonaws" % "aws-lambda-java-events" % "3.1.0"
   lazy val s3Sdk = "software.amazon.awssdk" % "s3" % "2.13.18"
-  lazy val ses = "software.amazon.awssdk" % "ses" % "2.15.2"
+  lazy val sesSdk = "software.amazon.awssdk" % "ses" % "2.15.2"
   lazy val sqsSdk = "software.amazon.awssdk" % "sqs" % "2.13.15"
   lazy val ecrSdk = "software.amazon.awssdk" % "ecr" % "2.13.15"
+  lazy val sfnSdk = "software.amazon.awssdk" % "sfn" % "2.15.79"
```
Comment (suzannehamilton) [inline project/Dependencies.scala]: Is there a particular reason for using different versions of all the AWS SDKs? Or could we have a variable for the version number so we can upgrade them all at once?

**Review by suzannehamilton — COMMENTED** [review]
Just one minor comment.

It'd also be good if @MancunianSam could take a look at this - you know more about this library and about Cats, Monix, etc.

File: project/Dependencies.scala (hunk 1 above)
Comment (TomJKing) [reply project/Dependencies.scala]: Nope, I just grabbed the latest version of the library from the maven repository. I'll put in a variable.

**Review by MancunianSam — APPROVED** [review]
Looks good, cheers.

**Pull request merged**